x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/connector/action/UpdateConnectorIndexNameAction.java

```diff
@@ -107,14 +109,12 @@ public ActionRequestValidationException validate() {
                 validationException = addValidationError("[connector_id] cannot be [null] or [\"\"].", validationException);
             }
 
-            if (Strings.isNullOrEmpty(indexName)) {
-                validationException = addValidationError("[index_name] cannot be [null] or [\"\"].", validationException);
-            }
-
-            try {
-                MetadataCreateIndexService.validateIndexOrAliasName(indexName, InvalidIndexNameException::new);
-            } catch (InvalidIndexNameException e) {
-                validationException = addValidationError(e.toString(), validationException);
+            if (indexName != null) {
+                try {
+                    MetadataCreateIndexService.validateIndexOrAliasName(indexName, InvalidIndexNameException::new);
+                } catch (InvalidIndexNameException e) {
+                    validationException = addValidationError(e.toString(), validationException);
+                }
```

Comment:
This could live inside some utils function as it's repeated three times, but nothing blocking